src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx

```diff
@@ -17,6 +21,22 @@ const Stack = createStackNavigator<OnboardingModalNavigatorParamList>();
 function OnboardingModalNavigator() {
     const styles = useThemeStyles();
     const {shouldUseNarrowLayout} = useOnboardingLayout();
+    const [hasCompletedGuidedSetupFlow] = useOnyx(ONYXKEYS.NVP_ONBOARDING, {selector: (onboarding) => onboarding?.hasCompletedGuidedSetupFlow ?? true});
+
+    useEffect(() => {
+        if (!hasCompletedGuidedSetupFlow) {
+            return;
+        }
+
+        // Need to go back to previous route and then redirect to Concierge,
+        // otherwise going back on concierge will go to onboarding and then redirected to concierge again
+        Navigation.goBack();
+        Report.navigateToConciergeChat();
```

Comment:
```suggestion
        Navigation.isNavigationReady().then(() => {
            Navigation.goBack();
            Report.navigateToConciergeChat();
        }
```

https://github.com/Expensify/App/assets/102477862/20c811c0-5999-4ace-b257-375c7946219b

I see the navigator is not ready when navigating to these routes. So, it goes back to onboarding again as seen in the video.

Reply:
I thought, this component is defined in Route, so navigation should be ready already...Pushed fix as you mentioned